### ISSUE TITLE
Rename `ZSH` variable to `OMZ` or similar

### ISSUE THREAD

**Issue by ghost** [ohmyzsh/ohmyzsh#10649]
### Describe the bug

I use the ZSH variable in my dotfiles and I can't use Oh My Zsh because it uses the same variable for a different folder. I'd expect the variable to be OMZ or something specific to the project.

**Comment by rwmitchell** [comment]
Agreed.  Oh My Zsh should use/set 'OMZ' (instead of ZSH) as that is the direct purpose it is being used and is much clearer.

I suspect any such change would require a long phase in to avoid breaking lots of stuff.

**Comment by rwmitchell** [comment]
however, it isn't likely to happen - you'll be better off changing your local usage of ZSH to something else.

**Comment by carlosala** [comment]
> however, it isn't likely to happen - you'll be better off changing your local usage of ZSH to something else.

I agree with you. OMZ defines itself as a framework, and not as a library. A framework conditions you to do some things at its way, in some aspects is more rigid than other options.
Renaming the variable would be an absolutely crazy breaking change, and would be really difficult to do a clean transition.

**Comment by robbyrussell** [comment]
Thanks for raising it, and sorry for the long silence.

Carlo's answer from 2022 still stands. `$ZSH` appears 117 times across 25 files here, `ZSH_CUSTOM` and `ZSH_CACHE_DIR` derive from it, and `export ZSH="$HOME/.oh-my-zsh"` is written into every user's `.zshrc` by our template — so every third-party plugin, theme and dotfiles repo referencing `$ZSH/custom` would break.

Adding `$OMZ` alongside doesn't help either: `oh-my-zsh.sh` would still read `$ZSH` and adopt whatever you'd set it to. Same reasoning we used closing #9543 on moving the install directory.

If you hit a real conflict, setting `ZSH` explicitly before the `source` line works — we honour it. Closing this out.
